File: AkkaWithZeroMQ/src/main/scala/org/akka/essentials/zeromq/example3/server/ServerActor.scala
package org.akka.essentials.zeromq.example3.server
import akka.zeromq.ZeroMQExtension
import akka.actor.ActorLogging
import akka.actor.Actor
import akka.zeromq.SocketType
import akka.zeromq.ZMQMessage
import akka.zeromq.Bind
import akka.zeromq.Listener
import akka.zeromq.Frame

class ServerActor extends Actor with ActorLogging {
  val repSocket = ZeroMQExtension(context.system).newSocket(SocketType.Rep, Bind("tcp://127.0.0.1:1234"), Listener(self))

  def receive: Receive = {
    case m: ZMQMessage =>
      var mesg = new String(m.payload(0));
      repSocket ! ZMQMessage(Seq(Frame(mesg + " Good to see you!")))
  }
}

File: AkkaWithZeroMQ/src/main/java/org/akka/essentials/zeromq/example3/server/ServerActor.java
package org.akka.essentials.zeromq.example3.server;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;
import akka.zeromq.Bind;
import akka.zeromq.Frame;
import akka.zeromq.Listener;
import akka.zeromq.SocketOption;
import akka.zeromq.ZMQMessage;
import akka.zeromq.ZeroMQExtension;

public class ServerActor extends UntypedActor {
	ActorRef repSocket = ZeroMQExtension.get(getContext().system())
			.newRepSocket(
					new SocketOption[] { new Bind("tcp://127.0.0.1:1237"),
							new Listener(getSelf()) });

	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof ZMQMessage) {
			ZMQMessage m = (ZMQMessage) message;
			String mesg = new String(m.payload(0));

			repSocket.tell((new ZMQMessage(
					new Frame(mesg + " Good to see you!"))));
		}
	}
}
